### ISSUE TITLE
Recent versions of buildkit won't reinstall old sites

### ISSUE THREAD

**Issue by MegaphoneJon** [civicrm/civicrm-buildkit#503]
I just ran `civibuild reinstall` and it failed with `missing variable: CMS_VERSION [in cvutil_inject_settings]`.  That seems to tied to the commits from 3 weeks ago.

I looked in `<civicrm-buildkit-root>/build/dmaster.sh` and saw `CMS_VERSION=""` which I changed to `CMS_VERSION="latest"` but that yielded this error:

```
+ php -d mysql.default_host=127.0.0.1:3306 -d mysql.default_user=dmasterciv_bkv7o -d mysql.default_password=CYlnSyGXHxLMPa4k GenCode.php schema/Schema.xml '' Drupal                                                                         
                                                                                                                                             
civicrm_domain.version := 5.23.alpha1                                                                                                                                                                                                        
                                                                                                                                                                                                                                            
Parsing schema description schema/Schema.xml                                                                                                                                                                                                
Extracting database information                                                                                                                                                                                                          
Extracting table information
PHP Fatal error:  Uncaught RuntimeException: Invalid configuration: the [cms.root] path must be an absolute URL in /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php:174
Stack trace:
#0 /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php(151): Civi\Core\Paths->toAbsoluteUrl('/', 'cms.root')
#1 /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php(176): Civi\Core\Paths->getVariable('cms.root', 'url')
#2 /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php(151): Civi\Core\Paths->toAbsoluteUrl('/', 'civicrm.root')
#3 /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php(224): Civi\Core\Paths->getVariable('civicrm.root', 'path')
#4 /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php(84): Civi\Core\Paths->getPath('l10n')
#5 [internal function]: Civi\Core\Paths->Civi\Core\{closure}()
#6 /home/jo in /home/jon/local/civicrm-buildkit/build/dmaster/sites/all/modules/civicrm/Civi/Core/Paths.php on line 174
```

Running `civibuild destroy` and recreating the site worked.

**Comment by seamuslee001** [comment]
@totten thoughts?

**Comment by totten** [comment]
It's odd that the `CMS_VERSION` would be missing/blank, but I suppose the condition ordinarily ought not be fatal, so I've posted a PR to relax it (#507). FWIW, in my `dmaster.sh` file, the value is `CMS_VERSION="7.x"`.

The other error... I don't think that's related to `CMS_VERSION` (merely obscured by it). I was able to reproduce the error -- remove `l10n` folder and ru `./bin/setup.sh -g`. Since the current practice in most `app/config/*/download.sh` scripts is to download l10n, we don't see the problem in CI . But it's always been valid to run GenCode without `l10n` data, so let me take a gander to see if there's an obvious resolution...
